tensorflow_quantum/core/ops/tfq_ps_weights_from_symbols_op.cc

```diff
@@ -146,7 +146,7 @@ class TfqPsWeightsFromSymbolOp : public tensorflow::OpKernel {
     auto DoWork2 = [&](int start, int end) {
       for (int i = start; i < end; i++) {
         for (int j = 0; j < n_symbols; j++) {
-          for (int k = 0; k < output_results.at(i).at(j).size(); k++) {
+          for (size_t k = 0; k < output_results.at(i).at(j).size(); k++) {
             output_tensor(i, j, k) = output_results.at(i).at(j).at(k);
           }
           for (int k = output_results.at(i).at(j).size();
```

Comment:
Why not here ?

Reply:
I agree this one looks inconsistent. It's because in this case, the value of k is being compared to `largest_single_symbol` on the next line, and that variable is an int.

The weird thing is k is initialized by setting its value to the output of `size()`, and yet the compiler isn't complaining about that part. I don't have a good explanation for that.